test/hotspot/jtreg/runtime/os/TestTracePageSizes.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=no-options
+ * @summary Run test with no arguments apart from the ones required by
+ *          the test.
+ * @requires os.family == "linux"
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log TestTracePageSizes
+ */
+
+/*
+ * @test id=large-page-size-1g
+ * @summary Run test with 1g pages on x64. Excluding ZGC since it fail
+ *          initialization if no large pages are available on the system.
+ * @requires os.family == "linux"
+ * @requires os.arch=="amd64" | os.arch=="x86_64"
+ * @requires vm.gc != "Z"
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseLargePages -XX:LargePageSizeInBytes=1g TestTracePageSizes
```

Comment:
Should we also test with -XX:LargePageSizeInBytes=2M?

Reply:
Good point, 2m might not always be the default so I guess it makes sense to explicitly test both.